### DISCUSSION TITLE
[question] Check if editor is empty

### DISCUSSION THREAD

**Discussion by timlkelly** [ueberdosis/tiptap#748] [General Questions]
Is it possible to check if the editor does not contain any user input? From what I see, there is always a root node with a paragraph element so it does not look like it is every truly "empty", but I'd like to check if there is no user input so I can disable/enable a button accordingly.

**Comment by mvind** [comment]
This is probably not the best solution but it is fast to implement. But you could simply check the length of the initial editor html string like 
```javascript
const content = this.editor.getHTML()
const empty_content = '<p></p>'
if(content.length == empty_content.length) {
    // editor is empty
}
```
Or you could compare the content. This is the way I have implemented a quick way to check whether an editor is empty or not.

Also looking at the placeholder plugin there is this ```is-editor-empty``` class that you also could look after, but I'm using a custom prosemirror document and can't figure out how to use the placeholder plugin on that custom document.

**Comment by jnguyen32** [comment]
This would be useful. When you try to implement the `Floating Menu` example from the tiptap website, it only works when it detects a ```<p><br></p>```. However, when the document itself is empty (e.g., a brand new document), the first elements are ```<div class="Prosemirror" ...><p><br></p></div>```

The issue arises with how the Floating Menu detects a new line and it alwalys looks for ```<p><br></p>``` and a couple other conditions. 

Why should the Floating Menu only work once there are inputs? Most people who start a new document want the new-line floating menu to drop in a Header or something.
